cg/src/domains/management/stats/stats.service.js

```diff
@@ -0,0 +1,43 @@
+const { db } = require('../../../db');
+
+class StatsService {
+  constructor(database = db) {
+    this.db = database;
+  }
+
+  async stats() {
+    const [result] = await db('subjects')
+      .select(db.raw('count(subjects.id) as consented'))
+      .join('subject_keys', 'subjects.id', 'subject_keys.subject_id');
+    const consentedSubjectCount = parseInt(result.consented, 10);
+
+    const [result2] = await db('subjects').select(db.raw('count(subjects.id) as total'));
+    const totalSubjectCount = parseInt(result2.total, 10);
```

Comment:
If there are no subjects I suspect there will be an error thrown on `result2.total`.

Reply:
Never mind – count will always return result even if zero.
Now there's a test for this :)